src/core.js

```diff
@@ -159,3 +149,8 @@ jQuery.fn.size = function() {
 	migrateWarn( "jQuery.fn.size() is deprecated; use the .length property" );
 	return this.length;
 };
+
+jQuery.parseJSON = function() {
+	migrateWarn( "jQuery.parseJSON is deprecated; use JSON.parse" );
```

Comment:
I am wondering how to do this. The most common reason for this seems to have been servers returning `Content-Type: application/x-json` which we then tried to parse and rightly died since it wasn't JSON. The new AJAX code uses `JSON.parse()` directly which will bring back that error. In any case, `jQuery.parseJSON()` was defined to return `null` on falsy arguments so I think we still have to have the `!json` check there even if we don't give an additional warning about it.